plugins/gradle/src/main/groovy/com/newrelic/agent/android/ClassTransformWrapperTask.groovy

```diff
@@ -59,8 +60,16 @@ abstract class ClassTransformWrapperTask extends DefaultTask {
                                 def jarEntry = new JarEntry(relativePath.replace(File.separatorChar, '/' as char))
                                 try {
                                     jarOutputStream.putNextEntry(jarEntry)
-                                    transformer.processClassBytes(classFile, fileInputStream).withCloseable {
-                                        jarOutputStream << it
+                                    try {
+                                        transformer.processClassBytes(classFile, fileInputStream).withCloseable {
+                                            jarOutputStream << it
+                                        }
+                                    } catch (RuntimeException re) {
+                                        logger.warn("[ClassTransformTask] encountered a RuntimeException: " + re.getMessage() + ", output the original jar file")
+                                        IOUtils.copy(fileInputStream, jarOutputStream)
```

Comment:
Just stream the input to the output: ```jarOutputStream << fileInputStream;```

Also, if there's a problem during the stream read, we may only get part of it back. ot much we can do other than buffer the class bytes.